### PR TITLE
[addons][pvr] cleanups to match more coming C++ interface style and fix duplicated queue and VFS types 

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This include some C++ rework changes, where also be can in independent. Again thought to reduce the others size and to have only changes related for C -> C++ parts there.

Commit 1:
------------------------------------------------
**Fix duplicated queue and VFS types**

Fixes Debian packaging's header-check failure

Commit 2:
------------------------------------------------
**[addons][pvr] separate menu hook calls and remove union use**

This is thought to make the calls cleaner, with the union before
was it always a bit ugly and unsafe.

Now have every type a own function where give his related data.
Further is on EPG menu hook the whole entry given.

Commit 3:
------------------------------------------------
**[addons][pvr] rename PVR_PROPERTIES to AddonProperties_PVR**

Done to match like on other addon types and to match his function tables.
Also is inside AddonInstance_PVR the "struct" before added to have "C"
conform.

Commit 4:
------------------------------------------------
**[addons][pvr] sort API function tables**

This is to have a better overview and with videoplayer parts on end.
If videoplayer inside is kicked out in future and done my inputstream
instance does it not break API.

Commit 5:
------------------------------------------------
**[addons][pvr] rename GetAddonCapabilities to GetCapabilities**

This is done to have his name more equal to the addon types and
prepared for coming C++ interface where included inside class.

Commit 6:
------------------------------------------------
**[addons][pvr] pass channel unique id to SignalStatus and GetDescrambleInfo**

This thought that addon identify his related channel, for the case it becomes
more as one time in future used. Further can it be also if only one identify
a related inputstream class where do his work.

Commit 7:
------------------------------------------------
**[addons][pvr] increase PVR API to 6.5.0**

## Motivation and Context
Come to end where the PVR addons soon on new interface and the system becomes cleaned up,

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h [hunk 1]
```diff
@@ -224,7 +201,7 @@ namespace ADDON
      * @param type The message type.
      * @param format The format of the message to pass to display in XBMC.
      */
-    void QueueNotification(const queue_msg_t type, const char *format, ... )
+    void QueueNotification(const QueueMsg type, const char *format, ... )
```
Comment (phunkyfish) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h:204]: Can or should this be a reference?

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h [hunk 2]
```diff
@@ -502,21 +502,23 @@ extern "C"
...
   /*!
    * Get the signal status of the stream that's currently open.
+   * @param channelUid Channel unique identifier
    * @param signalStatus The signal status.
    * @return PVR_ERROR_NO_ERROR if the signal status has been read successfully, false otherwise.
    * @remarks Optional, and only used if PVR_ADDON_CAPABILITIES::bHandlesInputStream or PVR_ADDON_CAPABILITIES::bHandlesDemuxing is set to true.
    *          Return PVR_ERROR_NOT_IMPLEMENTED if this add-on won't provide this function.
    */
-  PVR_ERROR SignalStatus(PVR_SIGNAL_STATUS& signalStatus);
+  PVR_ERROR SignalStatus(int channelUid, struct PVR_SIGNAL_STATUS* signalStatus);
```
Comment (phunkyfish) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:511]: is the struct keyword required? I don't see it elsewhere in the API?

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h [hunk 3]
```diff
@@ -502,21 +502,23 @@ extern "C"
...
   /*!
    * Get the descramble information of the stream that's currently open.
+   * @param channelUid Channel unique identifier
    * @param [out] descrambleInfo The descramble information.
    * @return PVR_ERROR_NO_ERROR if the descramble information has been read successfully, false otherwise.
    * @remarks Optional, and only used if PVR_ADDON_CAPABILITIES::bSupportsDescrambleInfo is set to true.
    *          Return PVR_ERROR_NOT_IMPLEMENTED if this add-on won't provide this function.
    */
-  PVR_ERROR GetDescrambleInfo(PVR_DESCRAMBLE_INFO* descrambleInfo);
+  PVR_ERROR GetDescrambleInfo(int channelUid, struct PVR_DESCRAMBLE_INFO* descrambleInfo);
```
Comment (phunkyfish) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:521]: is the struct keyword required? I don't see it elsewhere in the API?

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h (hunk 1 above)
Comment (phunkyfish) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h:204]: Also, `const char* format, ...`

Left align `*`

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h [hunk 4]
```diff
@@ -30,7 +30,7 @@ extern "C"
    * @return PVR_ERROR_NO_ERROR if the properties were fetched successfully.
    * @remarks Valid implementation required.
    */
-  PVR_ERROR GetAddonCapabilities(PVR_ADDON_CAPABILITIES *pCapabilities);
+  PVR_ERROR GetCapabilities(PVR_ADDON_CAPABILITIES *pCapabilities);
```
Comment (phunkyfish) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:33]: Left alight `*`

**Review by phunkyfish — CHANGES_REQUESTED** [review]

**Comment by phunkyfish** [comment]
@basilgello yours is the first commit in this PR.

**Comment by basilgello** [comment]
I noticed that @phunkyfish :) But thanks for the message!

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h:204]: Ah thanks, has not thought about :smile:

File: xbmc/pvr/addons/PVRClient.cpp [hunk 5]
```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
...
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_CHANNEL;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallChannelMenuHook(&menuHook, &tag);
```
Comment (ksooo) [inline xbmc/pvr/addons/PVRClient.cpp:1404]: Why do I have to set `menuHook.category`? The called method already defines the semantics, so no need to say which "category". It's clear that it must be "channel" when calling `CallChannelMenuHook`. Same comment applies to the other "Call*MenuHook" calls.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h (hunk 3 above)
Comment (AlwinEsch) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:521]: Yes and no, also forgot add on `PVR_ERROR (__cdecl* GetDescrambleInfo)(int, PVR_DESCRAMBLE_INFO*);`

As the addons use currently only C++ is it not required, but to have it as "C" only becomes it needed. But I remove from there and becomes done on rework request.

This request for here is mostly thought to reduce amount of changed places outside of PVR addon header.

File: xbmc/pvr/addons/PVRClient.cpp [hunk 6]
```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
...
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = bDeleted ? PVR_MENUHOOK_DELETED_RECORDING : PVR_MENUHOOK_RECORDING;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallRecordingMenuHook(&menuHook, &tag);
```
Comment (ksooo) [inline xbmc/pvr/addons/PVRClient.cpp:1419]: Looks like here comes `menuHook.category` (see my previous comment) into play, but why this inconsistency? Either we introduce `CallRecordingMenuHook` and `CallDeletedRecordingMenuHook` and get rid of `PVR_MENUHOOK::category` or ... as I don't get the benefit of this whole change to make 5 (6?) `Call*MenuHook` functions out of the one API call we had before ... just revert this whole change?

**Review by ksooo — COMMENTED** [review]
To be honest; I don't understand the ratio behind the menu hook related changes. They are not consistent (wrt. PVR_MENUHOOK::category usage), introduce 5 (6?) API functions where formerly one was sufficient - without any functional gain.

File: xbmc/pvr/addons/PVRClient.cpp (hunk 6 above)
Comment (AlwinEsch) [reply xbmc/pvr/addons/PVRClient.cpp]: Have this related commit removed and becomes done on C++ rework.

**Comment by AlwinEsch** [comment]
All addons are prepared for this request, OK to bring in (if jenkins have a happy day today)?

jenkins build this please

**Review by phunkyfish — APPROVED** [review]
Yes, fine by me.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by basilgello** [comment]
@AlwinEsch https://github.com/kodi-pvr/pvr.argustv/pull/103 still open.

**Pull request merged**